vcd/resource_vcd_external_network_v2.go

```diff
@@ -154,6 +154,18 @@ func resourceVcdExternalNetworkV2() *schema.Resource {
 				ForceNew:    true,
 				Description: "Enables IP Spaces for this network (default 'false'). VCD 10.4.1+",
 			},
+			"nat_and_firewall_service_intention": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Defines intentions to configure NAT and Firewall rules (only with IP Spaces, VCD 10.5.1+) One of `PROVIDER_GATEWAY`,`EDGE_GATEWAY`,`PROVIDER_AND_EDGE_GATEWAY`",
+			},
+			"route_advertisement_intention": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Defines intentions to configure route advertisement (only with IP Spaces, VCD 10.5.1+) One of `IP_SPACE_UPLINKS_ADVERTISED_STRICT`,`IP_SPACE_UPLINKS_ADVERTISED_FLEXIBLE`,`ALL_NETWORKS_ADVERTISED`",
```

Comment:
Shouldn't we have a validation function here?
Otherwise, we end up with a BAD_REQUEST error

Reply:
I added validations to both fields `nat_and_firewall_service_intention` and `route_advertisement_intention`. 

Test run worked:
```
=== RUN   TestAccVcdExternalNetworkV2Datasource
--- PASS: TestAccVcdExternalNetworkV2Datasource (16.87s)
=== RUN   TestAccVcdExternalNetworkV2NsxtVrf
--- PASS: TestAccVcdExternalNetworkV2NsxtVrf (81.89s)
=== RUN   TestAccVcdExternalNetworkV2NsxtT0Router
--- PASS: TestAccVcdExternalNetworkV2NsxtT0Router (79.04s)
=== RUN   TestAccVcdExternalNetworkV2Nsxv
--- PASS: TestAccVcdExternalNetworkV2Nsxv (44.48s)
=== RUN   TestAccVcdExternalNetworkV2NsxtSegment
--- PASS: TestAccVcdExternalNetworkV2NsxtSegment (31.78s)
=== RUN   TestAccVcdExternalNetworkV2NsxtConfigError
--- PASS: TestAccVcdExternalNetworkV2NsxtConfigError (1.67s)
=== RUN   TestAccVcdExternalNetworkV2NsxtSegmentIntegration
--- PASS: TestAccVcdExternalNetworkV2NsxtSegmentIntegration (102.35s)
=== RUN   TestAccVcdExternalNetworkV2NsxtIpSpace
--- PASS: TestAccVcdExternalNetworkV2NsxtIpSpace (97.09s)
=== RUN   TestAccVcdExternalNetworkV2NsxtTopologyIntentionEdgeAndStrict
--- PASS: TestAccVcdExternalNetworkV2NsxtTopologyIntentionEdgeAndStrict (35.87s)
=== RUN   TestAccVcdExternalNetworkV2NsxtTopologyIntentionProviderAndFlexible
--- PASS: TestAccVcdExternalNetworkV2NsxtTopologyIntentionProviderAndFlexible (37.07s)
=== RUN   TestAccVcdExternalNetworkV2NsxtTopologyIntentionProviderAndEdgeAllNetworks
--- PASS: TestAccVcdExternalNetworkV2NsxtTopologyIntentionProviderAndEdgeAllNetworks (36.98s)
=== RUN   TestAccVcdExternalNetworkV2NsxtTopologyIntentionIntegration
--- PASS: TestAccVcdExternalNetworkV2NsxtTopologyIntentionIntegration (157.57s)
```